lib/identify.go

```diff
@@ -0,0 +1,69 @@
+package vole
+
+import (
+	"context"
+	"sort"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/peerstore"
+	"github.com/libp2p/go-libp2p/core/protocol"
+	"github.com/multiformats/go-multiaddr"
+)
+
+type IdentifyInfo struct {
+	ProtocolVersion string
+	AgentVersion    string
+	Addresses       []multiaddr.Multiaddr
+	Protocols       []protocol.ID
+}
+
+func IdentifyRequest(ctx context.Context, maStr string) (*IdentifyInfo, error) {
+	ai, err := peer.AddrInfoFromString(maStr)
+	if err != nil {
+		return nil, err
+	}
+
+	h, err := libp2pHost()
+	if err != nil {
+		return nil, err
+	}
+
+	if err := h.Connect(ctx, *ai); err != nil {
+		return nil, err
+	}
+
```

Comment:
Don't you need to call `IdentifyWait` here?

Reply:
Shouldn't that be covered by the BasicHost for me? https://github.com/libp2p/go-libp2p/blob/d3004eaf3cfe2388a39cd0e1bfa76da2153cfe50/p2p/host/basic/basic_host.go#L723-L759

Reply:
You're right.